### PR TITLE
New Feature: ArrayConvertingPopulator

### PR DESCRIPTION
**Pull request by mike4git**
In addition to the ConvertingPopulator I would like to offer an ArrayConvertingPopulaotr doing the same thing for arrays of objects.

### CODE REVIEW

File: src/Populator/ArrayConvertingPopulator.php
Comment (nsd0sneumann) [inline src/Populator/ArrayConvertingPopulator.php:1]: wie zusammen ausgearbeitet:
```php
<?php

declare(strict_types=1);

namespace Neusta\ConverterBundle\Populator;

use Neusta\ConverterBundle\Converter;
use Neusta\ConverterBundle\Exception\PopulationException;
use Neusta\ConverterBundle\Populator;
use Symfony\Component\PropertyAccess\PropertyAccessorInterface;

/**
 * A populator that uses a converter to convert a field of type array<TInnerSource> from TSource
 * into an object of type array<TInnerTarget> for a field of TTarget.
 *
 * @template TSource of object
 * @template TTarget of object
 * @template TContext of object|null
 *
 * @implements Populator<TSource, TTarget, TContext>
 */
final class ArrayConvertingPopulator implements Populator
{
    private ArrayPropertyMappingPopulator $populator;

    /**
     * @template TInnerSource of object
     * @template TInnerTarget of object
     *
     * @param Converter<TInnerSource, TInnerTarget, TContext> $converter
     */
    public function __construct(
        Converter $converter,
        string $sourcePropertyName,
        string $sourcePropertyName2,
        string $targetPropertyName,
        PropertyAccessorInterface $itemAccessor = null,
        PropertyAccessorInterface $accessor = null,
    ) {
        $this->populator = new ArrayPropertyMappingPopulator(
            $targetPropertyName,
            $sourcePropertyName,
            $sourcePropertyName2,
            \Closure::fromCallable([$converter, 'convert']),
            $itemAccessor,
            $accessor,
        );
    }

    /**
     * @throws PopulationException
     */
    public function populate(object $target, object $source, ?object $ctx = null): void
    {
        $this->populator->populate($target, $source, $ctx);
    }
}```

File: src/Populator/ArrayPropertyMappingPopulator.php
Comment (nsd0sneumann) [inline src/Populator/ArrayPropertyMappingPopulator.php:1]: as worked out together:
```php
<?php

declare(strict_types=1);

namespace Neusta\ConverterBundle\Populator;

use Neusta\ConverterBundle\Exception\PopulationException;
use Neusta\ConverterBundle\Populator;
use Symfony\Component\PropertyAccess\PropertyAccess;
use Symfony\Component\PropertyAccess\PropertyAccessorInterface;

/**
 * @template TSource of object
 * @template TTarget of object
 * @template TContext of object|null
 *
 * @implements Populator<TSource, TTarget, TContext>
 */
final class ArrayPropertyMappingPopulator implements Populator
{
    /** @var \Closure(mixed, TContext=):mixed */
    private \Closure $mapper;
    private PropertyAccessorInterface $itemAccessor;
    private PropertyAccessorInterface $accessor;

    /**
     * @param \Closure(mixed, TContext=):mixed|null $mapper
     */
    public function __construct(
        private string            $targetProperty,
        private string            $sourceProperty,
        private string            $sourceProperty2,
        ?\Closure                 $mapper = null,
        PropertyAccessorInterface $itemAccessor = null,
        PropertyAccessorInterface $accessor = null,
    )
    {
        $this->mapper = $mapper ?? static fn($v) => $v;
        $this->itemAccessor = $itemAccessor ?? PropertyAccess::createPropertyAccessor();
        $this->accessor = $accessor ?? PropertyAccess::createPropertyAccessor();
    }

    /**
     * @throws PopulationException
     */
    public function populate(object $target, object $source, ?object $ctx = null): void
    {
        try {
            $unwrappedArray = array_map(
                function ($arrayItem) use ($ctx) {
                    return $this->itemAccessor->getValue($arrayItem, $this->sourceProperty2);
                },
                $this->accessor->getValue($source, $this->sourceProperty)
            );


            $this->accessor->setValue(
                $target,
                $this->targetProperty,
                array_map(
                    function ($item) use ($ctx) {
                        return ($this->mapper)($item, $ctx);
                    },
                    $unwrappedArray
                )
            );
        } catch (\Throwable $exception) {
            throw new PopulationException($this->sourceProperty, $this->targetProperty, $exception);
        }
    }

}
```

File: src/Populator/ArrayConvertingPopulator.php
Comment (nsd0sneumann) [reply src/Populator/ArrayConvertingPopulator.php:1]: Mir ist noch was eingefallen: Es muss möglich sein 1-x Arrays bei sourceProperty zu übergeben

Comment (mike4git) [reply src/Populator/ArrayConvertingPopulator.php:1]: Why?
Could you please make it clearer what's your purpose for this change?!

Comment (mike4git) [reply src/Populator/ArrayConvertingPopulator.php:1]: Additionally this PR workflow fails - and I don't know why or how to readjust?!

Comment (nsd0sneumann) [reply src/Populator/ArrayConvertingPopulator.php:1]: > Why? Could you please make it clearer what's your purpose for this change?!

my mistake, I had thought of a specific use case. But then it is converted differently

File: docs/usage.md [hunk 1]
```diff
@@ -164,6 +164,160 @@ $person = $this->converter->convert($user);
 
 Conversion done.
 
+## Special Populators
```
Comment (lukadschaak) [inline docs/usage.md:200]: Here are the docs from the other PR included, right? So we have to merge this first and then rebase this one here correctly.

**Pull request merged**